extendr-api/src/wrapper/nullable.rs

```diff
@@ -77,3 +131,88 @@ where
         }
     }
 }
+
+impl<T> Into<Option<T>> for Nullable<T> {
+    /// Convert a Nullable type into Option
+    /// ```
+    /// use extendr_api::prelude::*;
+    /// test! {
+    ///     assert_eq!(Into::<Option<i32>>::into(Nullable::Null), None);
+    ///     assert_eq!(Into::<Option<i32>>::into(Nullable::NotNull(42)), Some(42));
+    /// }
+    /// ```
+    fn into(self) -> Option<T> {
+        match self {
+            Nullable::NotNull(value) => Some(value),
+            _ => None,
+        }
+    }
+}
+
+impl<'a, T> Into<Option<&'a T>> for &'a Nullable<T> {
+    /// Convert a Nullable type into Option
+    /// ```
+    /// use extendr_api::prelude::*;
+    /// test! {
+    ///     assert_eq!(Into::<Option<&i32>>::into(&Nullable::Null), None);
+    ///     assert_eq!(Into::<Option<&i32>>::into(&Nullable::NotNull(42)), Some(&42));
+    /// }
+    /// ```
+    fn into(self) -> Option<&'a T> {
+        match self {
```

Comment:
There's some redundancy here. Should we either a) delegate some of these impls: https://stackoverflow.com/a/56567737/2148718 or b) solve it with a macro (https://github.com/auto-impl-rs/auto_impl/). 

Because you haven't considered `mut` references here either.

Reply:
Maybe it is an argument for dropping `Option` here altogether and instead borrowing some methods from `Option` itself?

Reply:
Interesting idea, but I don't think that would be any cleaner. I think that might actually result in more code and redundancy rather than less.

Reply:
I agree that it doesn't make much sense to take a `&mut Nullable<T>` because there's no actual mutable state there. But on the other hand, there's no reason we can't convert that into an `Option`. It's just a reference with my permissions than usual.

Reply:
So, right now we have no constraints on `T` when implementing `Nulalble<T> <--> Option<T>`. In reality though, `Nullable<T>` can be only constructed from something that can be `TryFrom<RObj>` or `Into<Robj>` (or some variation). I'd say we could put some trait constraints on `T` everywhere (something that says `T` should be `Robj` - compatible), in which case I see no real reason to expose `mut` reference support, since there is no scenario where we allow such things.

Reply:
Fair enough, there is no practical use case for converting mutable references.

But on that note, I like your idea for locking down the `impl` for `Nullable` using trait bounds to enforce that it's always a `Nullable<T> where T: TryFrom<Robj>`. I like the clear message of putting the trait bounds on the actual `enum` itself, but because https://github.com/rust-lang/rust/issues/44491 is still open, I guess it's best to just put the bounds on the `impl`.

Reply:
So if you convert from `Nullable<T>` to `Option<T>`, we should require `T : TryFrom<Robj>`, it is a scenario for accepting a `Nullable<T>`. However, if we convert from `Option<T>` to `Nullable<T>`, we require `Into<Robj>` since it is a scenario of returning a `Nullable<T>`. The majority of types implement both traits, but not e.g. `ndarray.`
So we should be able to go `Nullable<ArrayView>` -> `Option<ArrayView>`, and then take an `Option<ArrayBase>` and convert to `Nullable<ArrayBase>` to return it back as nullable.